pkg/ebpf/c/common/filtering.h

```diff
@@ -434,8 +437,11 @@ statfunc u64 match_data_filters(program_data_t *p, u8 index)
     // 2. Default Enabled Policies: Policies that are enabled by default (default_enable_policies)
     // remain enabled only if they are not explicitly disabled (explicit_disable_policies).
     res = explicit_enable_policies | (default_enable_policies & ~explicit_disable_policies);
+    // Combine policies that use string filters with those that do not
```

Comment:
NOTE: Not directly related to this PR, but I noticed that the logic on lines 384-386, 407-409, and 428-430 could be refactored into a helper function for better reusability and clarity.

In fact, it might be worth considering whether the [entire type match logic](https://github.com/aquasecurity/tracee/pull/4468/files#diff-c046893996f2e2a6e44a8caac7a14abbf47b47486ee5ceaa9029c34fc5f94a99R389-R410) (prefix, suffix) could be consolidated into a single helper function to streamline the code further.

Reply:
thank you to point that... I think its possible to do some refactor for clarity... specially for each filter ending logic where there are some duplicate logic. The next phase (https://github.com/aquasecurity/tracee/issues/4433), we'll expend to other type of filters, so I'll review the string filter logic... WDYT?

Reply:
Awesome, ping me then so we can discuss further.